plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -761,12 +763,85 @@ public void testBigQueryExternalTable()
+            assertQuery("SELECT table_type FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + biglakeExternalTable + "'", "VALUES 'BASE TABLE'");
+
+            assertThat(query("DESCRIBE test." + biglakeExternalTable)).matches("DESCRIBE tpch.region");
+            assertThat(query("SELECT * FROM test." + biglakeExternalTable)).matches("SELECT * FROM tpch.region");
+
+            assertUpdate("DROP TABLE test." + biglakeExternalTable);
+            assertQueryReturnsEmptyResult("SELECT * FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + biglakeExternalTable + "'");
+
+            // BigLake tables should not run queries, since they are read directly using the storage read API.
+            verifyThatStreamApiWasUsed(biglakeExternalTable);
+        }
+        finally {
+            onBigQuery("DROP EXTERNAL TABLE IF EXISTS test." + biglakeExternalTable);
+        }
+    }
+
+    @Test
+    public void testExternalObjectTable()
+    {
+        String objectTableConnectionId = bigQueryConnectionId;
+
+        String objectExternalTable = "test_object_external" + randomNameSuffix();
+
+        try {
+            onBigQuery("CREATE EXTERNAL TABLE test." + objectExternalTable +
+                       " WITH CONNECTION `" + objectTableConnectionId + "`" +
+                       " OPTIONS (object_metadata = 'SIMPLE', uris = ['gs://" + gcpStorageBucket + "/tpch/tiny/region.csv'])");
+            assertQuery("SELECT table_type FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + objectExternalTable + "'", "VALUES 'BASE TABLE'");
+
+            assertThat(query("SELECT uri FROM test." + objectExternalTable)).succeeds();
+
+            assertUpdate("DROP TABLE test." + objectExternalTable);
+            assertQueryReturnsEmptyResult("SELECT * FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + objectExternalTable + "'");
+
+            // BigLake tables should not run queries, since they are read directly using the storage read API.
+            verifyThatStreamApiWasNotUsed(objectExternalTable);
+        }
+        finally {
+            onBigQuery("DROP EXTERNAL TABLE IF EXISTS test." + objectExternalTable);
+        }
+    }
+
+    private void verifyThatStreamApiWasUsed(String externalTable)
+    {
+        verifyHowManyTimesTableWasReferencedInJobs(externalTable, 0);
+    }
+
+    private void verifyThatStreamApiWasNotUsed(String externalTable)
+    {
+        verifyHowManyTimesTableWasReferencedInJobs(externalTable, 1);
+    }
+
+    private void verifyHowManyTimesTableWasReferencedInJobs(String externalTable, long noOfReferences)
+    {
+        assertThat(bigQuerySqlExecutor.executeQuery("""
+                 SELECT count(*) FROM region-us.INFORMATION_SCHEMA.JOBS WHERE EXISTS(
+                     SELECT * FROM UNNEST(referenced_tables) AS referenced_table
+                         WHERE referenced_table.table_id = '%s')
```

Comment:
We faced flaky issues in `assertLabelForTable` using `NFORMATION_SCHEMA.JOBS_BY_USER` table. Does this query return the latest status without delay?

Reply:
I did not noticed locally, however I added eventual assert

Reply:
We shouldn't use eventual assertion if there is no delay. 

We can't recognize whether the reference count was exactly 0 or delay happened in the following assertion:
```sql
assertEventually(() -> assertThat(getTableReferenceCountInJob(biglakeExternalTable)).isEqualTo(0));
```

Reply:
The only idea that comes to my mind is periodical check with timeout. I don't like sleep in the code but if we expect delays so the sleep usage seems to be the one of the simplest solution.